blazor/gantt-chart/virtualization.md

```diff
@@ -13,159 +13,118 @@ Gantt Chart allows you to load a large amount of data without performance degrad
+        this.TaskCollection = VirtualData.GetTreeVirtualData(1000);
+    }
+    
+    public class VirtualData
+    {
+        public static List<TaskData> GetTreeVirtualData(int count)
         {
-            List<GanttDataSource> list = new List<GanttDataSource>();
-            for (var i = 0; i < 2; i++)
+            List<TaskData> DataCollection = new List<TaskData>();
+            Random rand = new Random();
+            var x = 0;
+            int duration = 0;
+            DateTime startDate = new DateTime(2000, 1, 5);
+            DateTime endDate = new DateTime(2000, 1, 12);
+            string[] assignee = { "Allison Janney", "Bryan Fogel", "Richard King", "Alex Gibson" };
+            string[] reporter = { "James Ivory", "Jordan Peele", "Guillermo del Toro", "Gary Oldman" };
+            for (var i = 1; i <= count / 5; i++)
             {
-                var x = list.Count + 1;
-                list.Add(new GanttDataSource()
-                {
-                    TaskId = x,
-                    TaskName = "Project " + (i + 1)
-                });
-                for (var j = 0; j < TempData().ToList().Count; j++)
-                {
-                    list.Add(new GanttDataSource()
+                var name = rand.Next(0, 100);
+                TaskData Parent = new TaskData()
                     {
-                        TaskId = TempData()[j].TaskId + x,
-                        TaskName = TempData()[j].TaskName,
-                        StartDate = TempData()[j].StartDate,
-                        Duration = TempData()[j].Duration,
-                        Progress = TempData()[j].Progress,
-                        ParentID = TempData()[j].ParentID + x,
-                        Predecessor = TempData()[j].Predecessor
-                    });
+                        ID = ++x,
+                        TaskName = "Task " + x,
+                        StartDate = startDate,
+                        EndDate = startDate.AddDays(26),
+                        Duration = "20",
+                        Assignee = "Mark Bridges",
+                        Reporter = "Kobe Bryant",
+                        Progress = rand.Next(100),
+                    };
+                DataCollection.Add(Parent);
+                for (var j = 1; j <= 4; j++)
+                {
+                    startDate = startDate.AddDays(j == 1 ? 0 : duration + 2);
+                    duration = 5;
+                    DataCollection.Add(new TaskData()
+                        {
+                            ID = ++x,
+                            TaskName = "Task " + x,
+                            StartDate = startDate,
+                            EndDate = startDate.AddDays(5),
+                            Duration = duration.ToString(),
+                            Assignee = assignee[j - 1],
+                            Reporter = reporter[j - 1],
+                            Progress = rand.Next(100),
```

Comment:
Don't use random in UG code snippets. It will raise issue in UG automation.

Reply:
ok